### PR TITLE
feat: allow to enable strict json deserializer

### PR DESCRIPTION
**Pull request by simPod**
| Q             | A
| ------------- | ---
| New feature?  | yes
| Doc updated   | yes
| BC breaks?    | no 
| Deprecations? | no 
| Fixed tickets | Implements support for https://github.com/schmittjoh/serializer/pull/1401
| License       | MIT

### CODE REVIEW

**Review by scyzoryck — APPROVED** [review]

File: Resources/doc/configuration.rst [hunk 1]
```diff
@@ -241,6 +241,7 @@ values:
                     depth: 512
                 json_deserialization:
                     options: 0 # json_decode options bitmask
+                    strict: true
```
Comment (goetas) [inline Resources/doc/configuration.rst:244]: i think that the default value is `false`.

Comment (simPod) [reply Resources/doc/configuration.rst:244]:
```suggestion
                    strict: false
```

**Comment by simPod** [comment]
done

**Comment by goetas** [comment]
great, thanks!

**Comment by Arthur-GP** [comment]
Hey ! 
Problably a dumb question and might not be the good place to ask this.
But is there any documentation about this ? 
I can't find any either in the [serializer](http://jmsyst.com/libs/serializer#documentation)  or the [JMSSerializerBundle](http://jmsyst.com/bundles/JMSSerializerBundle)

**Comment by scyzoryck** [comment]
@Arthur-GP Unfortunately there is no docs yet. :( 
I believe it is super simple to configure - see `strict` option in bundle configuration. :) 

```
 php bin/console debug:config jms_serializer

Current configuration for extension with alias "jms_serializer"
===============================================================

jms_serializer:
    visitors:
        xml_serialization:
            format_output: true
            default_root_ns: ''
        json_serialization:
            options: 1216
        json_deserialization:
            options: 0
            strict: false
```

**Comment by Arthur-GP** [comment]
Yea well, that's what i did and it didn't work.

`HomeController.php`
```php
use JMS\Serializer\SerializerInterface;

class HomeController extends AbstractController
{
    private SerializerInterface $_serializer;

    public function __construct(SerializerInterface $serializer)
    {
        $this->_serializer = $serializer;
        
    }
    
    #[Route('/', name: 'home')]
    public function index(Request $request, KernelInterface $kernel): Response
    {
        $webService = new MoodleWebService($this->_serializer ,"https://".$listOfAllServerForNavBar[0]['nameserver']."/moodle",$list_service_for_server);
        $APIMoodleResult = $webService->getUser_moodle($current_user);

```
`MoodleWebService.php`
```php
class MoodleWebService
{

    private SerializerInterface $_serializer;
    private ?string $domainnename;
    private ?array $list_token_function;

    public function __construct(SerializerInterface $serializer, string $domainnename, array $list_token_function)
    {
        $this->domainnename = $domainnename;
        $this->list_token_function = $list_token_function;
        $this->_serializer = $serializer;
    }

    public function getUser_moodle(User $current_user)
    {
        $curl = curl_init();
        // Setup curl, like ssl, post, data send etc..
        $result = curl_exec($curl);
        if (curl_errno($curl)) {
            echo curl_error($curl);
            throw new Exception(curl_error($curl));
        }
        try {
            $APIMoodleResult = $this->_serializer->deserialize(
                $result,
                APIMoodleUser::class,
                'json',
            );
        } catch (Exception $e) {
            throw $e;
        }
        
        var_dump($APIMooldeResult);
        // object(App\Entity\MoodleAPI\APIMoodleUser)#596 (18) { ["id":"App\Entity\MoodleAPI\APIMoodleUser":private]=> NULL ["username":"App\Entity\MoodleAPI\APIMoodleUser":private]=> NULL ["firstname":"App\Entity\MoodleAPI\APIMoodleUser":private]=> NULL ["lastname":"App\Entity\MoodleAPI\APIMoodleUser":private]=> NULL ["fullname":"App\Entity\MoodleAPI\APIMoodleUser":private]=> NULL 
```

`config/packages/jms_serializer.yaml`
```yaml
jms_serializer:
    visitors:
        json_deserialization:
            strict: true
            
# Also try with default context
    default_context:
        deserialization:
            serialize_null: false
        serialization:
            serialize_null: false
```

`APIMoodleUser.php`
```php
use JMS\Serializer\Annotation as Serializer;
class APIMoodleUser
{

    #[Serializer\Type("int")]
    private $id; 
    
    #[Serializer\Type("string")]
    private $username; 
    
    // pretty much the same for other field
    // With getter & setter
```
I really don't understand why it doesn't work. It's been about 3 days and I'm starting to lose my mind 🙃
Perhaps it has something to do with how i create my serializer i guess ? I don't know.

**Comment by scyzoryck** [comment]
@Arthur-GP Have you managed to solve it? If not - could you provide some small repo with repro? :)

**Comment by Arthur-GP** [comment]
No, I didn't manage to solve it. All the code needed to produce this bug is above. I can make a repo if you're having trouble recreating this bug with just the code above.

I didn't close some bracket of the class because i wanted to be more readable

`src/Entity/APIMoodleUser.php` with the code above, and getter & setter too. 
`src/Controller/HomeController.php` with the code above
`src/Service/MoodleWebService.php` with the code above.
`config/packages/jms_serializer.yaml` with the param above.
`"jms/serializer-bundle": "^5.2",`

**Pull request merged**